components/ai_chat/renderer/page_content_extractor.cc

```diff
@@ -117,9 +102,43 @@ void PageContentExtractor::OnDestruct() {
   delete this;
 }
 
+base::WeakPtr<PageContentExtractor> PageContentExtractor::GetWeakPtr() {
+  return weak_ptr_factory_.GetWeakPtr();
+}
+
 void PageContentExtractor::ExtractPageContent(
     mojom::PageContentExtractor::ExtractPageContentCallback callback) {
   VLOG(1) << "AI Chat renderer has been asked for page content.";
+  // When content has been pushed to this class from a throttle via
+  // OnInterceptedPageContentChanged, use that content instead of fetching it
+  // from the page.
+  if (intercepted_content_) {
+    auto intercepted_content = std::move(intercepted_content_);
```

Comment:
what is this type?

Reply:
same as `intercepted_content_` which you also need to look where it's defined.